### DISCUSSION TITLE
Various errors trying to install Firedrake [macOS]

### DISCUSSION THREAD

**Discussion by tom-new** [firedrakeproject/firedrake#3108] [Firedrake support]
After a Homebrew update broke my Firedrake installation, I attempted to reinstall and was met with the following error:

```
File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/Users/tom/.virtualenvs/firedrake/bin/python3.11', '-m', 'ensurepip', '--upgrade', '--default-pip']' died with <Signals.SIGABRT: 6>.
```

Unable to resolve this, and at the recommendation of Steph at the G-ADOPT workshop just last week and David Ham on a previous post, I decided to try and use `pyenv` to avoid the issue of Homebrew updating Python 3 against my will. In order to start on a completely clean state, I reinstalled macOS, then installed Homebrew, and `pyenv` using Homebrew. When I tried to install Firedrake again, with Python 3.11.5 with pyenv, I was met with the same error message as in [this](https://github.com/firedrakeproject/firedrake/discussions/3090) discussion, and examining the install log it also appeared `islpy` was causing the issue. However, when I ran the command `python -m pip install --no-binary islpy -vvv islpy` in a clean virtual environment, `islpy` installed without issue. Then, when I tried to install firedrake again, the installation completed 'sucessfully', however when I tried to run the tests it threw a whole bunch of errors complaining that `islpy` could not be found.

I decided to try 'from scratch' again, and reinstalled it following [the instructions on the pyenv GitHub page](https://github.com/pyenv/pyenv#uninstalling-pyenv). Now, I get the same error as I did before I even reinstalled macOS and I am at my wits' end. Any help would be appreciated.

[firedrake-install.log](https://github.com/firedrakeproject/firedrake/files/12654698/firedrake-install.log)

**Comment by colinjcotter** [comment]
I can't remember how to do it, but you can ask Homebrew to install multiple versions of python. Then you can just use the path to the version you want to use to build the Firedrake venv.

________________________________________
From: Tom New ***@***.***>
Sent: 19 September 2023 02:30
To: firedrakeproject/firedrake
Cc: Subscribed
Subject: [firedrakeproject/firedrake] Various errors trying to install Firedrake [macOS] (Discussion #3108)

After a Homebrew update broke my Firedrake installation, I attempted to reinstall and was met with the following error:

File ***@***.***/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/Users/tom/.virtualenvs/firedrake/bin/python3.11', '-m', 'ensurepip', '--upgrade', '--default-pip']' died with <Signals.SIGABRT: 6>.


Unable to resolve this, and at the recommendation of Steph at the G-ADOPT workshop just last week and David Ham on a previous post, I decided to try and use pyenv to avoid the issue of Homebrew updating Python 3 against my will. In order to start on a completely clean state, I reinstalled macOS, then installed Homebrew, and pyenv using Homebrew. When I tried to install Firedrake again, with Python 3.11.5 with pyenv, I was met with the same error message as in this<https://github.com/firedrakeproject/firedrake/discussions/3090> discussion, and examing the install log it also appeared islpy was causing the issue. However, when I ran the command python -m pip install --no-binary islpy -vvv islpy in a clean virtual environment, islpy installed without issue. Then, when I tried to install firedrake again, the installation completed 'sucessfully', however when I tried to run the tests it threw a whole bunch of errors complaining that islpy could not be found.

I decided to try 'from scratch' again, and reinstalled it following the instructions on the pyenv GitHub page<https://github.com/pyenv/pyenv#uninstalling-pyenv>. Now, I get the same error as I did before I even reinstalled macOS and I am at my wits' end. Any help would be appreciated.

Uploading firedrake-install.log…<firedrake-install.log>

**Comment by dham** [comment]
Islpy broke for us last week while we were at the Firedrake meeting.  @JDBetteridge is fixing it in #3106. Please try using the install script from that branch: `https://raw.githubusercontent.com/firedrakeproject/firedrake/JDBetteridge/installation_hotfix/scripts/firedrake-install`

**Comment by tom-new** [comment]
This link seems broken—I assume this is because (as I see from the issue page) that the fixed branch has been merged into master?

**Reply by dham** [reply]
That is correct
